dvc/fs/utils.py

```diff
@@ -1,20 +1,37 @@
 import errno
 import logging
 import os
-from typing import TYPE_CHECKING, List, Optional
+from typing import IO, TYPE_CHECKING, List, Optional, Union, cast
 
+from ..utils.fs import as_atomic
+from ._callback import DEFAULT_CALLBACK, FsspecCallback
 from .base import RemoteActionNotImplemented
 from .local import LocalFileSystem
 
 if TYPE_CHECKING:
-    from .base import FileSystem
-    from .types import AnyPath
+    from dvc.types import AnyPath
+
+    from .base import AnyFSPath, FileSystem
 
 logger = logging.getLogger(__name__)
 
 
+def upload(
```

Comment:
```suggestion
def atomic_upload(
```

?

Reply:
I added `atomic=True/False` flag to it for now, so it's no longer `atomic` by default. So I think I'll just name it `upload` for now..